llvm-bazel/llvm-project-overlay/llvm/BUILD

```diff
@@ -2049,16 +2049,26 @@ cc_library(
     ],
 )
 
+# There is some deep layering violation between these libraries and JITLink.
+# To workaround this, the same headers are included in multiple rules.
+cc_library(
+    name = "Orc_layering_violation_headers",
+    hdrs = glob([
+        "include/llvm/ExecutionEngine/Orc/*.h",
+        "include/llvm/ExecutionEngine/Orc/RPC/*.h",
+        "include/llvm/ExecutionEngine/JITLink/*.h",
+    ]),
+)
+
 cc_library(
     name = "OrcJIT",
     srcs = glob([
         "lib/ExecutionEngine/Orc/*.cpp",
         "lib/ExecutionEngine/Orc/*.h",
-        "lib/ExecutionEngine/Orc/RPC/*.cpp",
-        "lib/ExecutionEngine/OrcError/*.cpp",
     ]),
     hdrs = glob([
         "include/llvm/ExecutionEngine/Orc/*.h",
+        "include/llvm/ExecutionEngine/Orc/RPC/*.h",
```

Comment:
Is this needed? I'm surprised that you took away `srcs` but need to add `hdrs`

Reply:
Well the srcs paths don't exist anymore, which is why I deleted them. This isn't strictly necessary because it ends up getting these headers from `Orc_layering_violation_headers` via `OrcShared` and `OrcTargetProcess` and we don't have `layering_check` turned on, but there are direct includes between `Orc/` `Orc/RPC` going both ways. I think this is the rule these belong in.